Plans: fix credit notice spacing

ciprianimike opened the pull request:
#### Proposed Changes

Addresses 820-gh-Automattic/martech

* Branch `update/plan-overhaul-revert-fix-spacing` is cut from `update/plan-overhaul-revert`
* Update `classNames` adding is-wpcom conditionally
* Use `is-wpcom` `className` to fix the spacing betweem the `Credits Notice` and the `Non-renewable Plan Notice` 

#### Testing Instructions

* Add (or be sure to have) WPCom credits in your WPCom account [https://wordpress.com/wp-admin/network/admin.php?page=store-admin](https://wordpress.com/wp-admin/network/admin.php?page=store-admin
* Create (or select) a `Pro plan` or `Starter plan` site
* Visit http://calypso.localhost:3000/plans/<YOUR SITE>
* Verify the gap between the Credits Notice and the Non-renewable Notice is resolved.

#### Before / After

##### Before
<img width="1478" alt="Screenshot 2022-07-18 at 15 57 15" src="https://user-images.githubusercontent.com/5706607/179565077-35a5747f-73f7-4401-9d78-94affa629f4d.png">

##### After
<img width="1459" alt="Screenshot 2022-07-18 at 19 06 36" src="https://user-images.githubusercontent.com/5706607/179565122-e17b2df4-18bc-4cd7-b65b-390da7ad4d8f.png">

claudiucelfilip committed:
Disable Pro and Starter config flags

claudiucelfilip committed:
Disable Pro and Starter config flags

claudiucelfilip committed:
Disable Pro and Starter config flags

claudiucelfilip committed:
Disable Pro and Starter config flags

gmovr committed:
Update link in UpsellNudge (#64496)

* Updating the CTA urls to go to the plans page, as they used to do

* Updating the specific UpsellNudge in the Settings > General form

* Updating a test assertion for form-general.jsx to expect Business

gmovr committed:
Update link in UpsellNudge (#64496)

* Updating the CTA urls to go to the plans page, as they used to do

* Updating the specific UpsellNudge in the Settings > General form

* Updating a test assertion for form-general.jsx to expect Business

gmovr committed:
Update link in UpsellNudge (#64496)

* Updating the CTA urls to go to the plans page, as they used to do

* Updating the specific UpsellNudge in the Settings > General form

* Updating a test assertion for form-general.jsx to expect Business

claudiucelfilip committed:
Remove tos-related files

claudiucelfilip committed:
Remove tos-related files

claudiucelfilip committed:
Remove tos-related files

claudiucelfilip committed:
Remove tos-related files

claudiucelfilip committed:
Skip Free to Pro test suite

claudiucelfilip committed:
Skip Free to Pro test suite

claudiucelfilip committed:
Skip Free to Pro test suite

claudiucelfilip committed:
Skip Free to Pro test suite

claudiucelfilip committed:
User better method to disable Free to Pro upgrade test

claudiucelfilip committed:
User better method to disable Free to Pro upgrade test

claudiucelfilip committed:
User better method to disable Free to Pro upgrade test

claudiucelfilip committed:
User better method to disable Free to Pro upgrade test

Commit:
fix credit notice spacing

Commit:
fix credit notice spacing

dzver commented on client/my-sites/plans-features-main/index.jsx:
```diff
@@ -459,7 +467,12 @@ export class PlansFeaturesMain extends Component {
+				<div
+					className={ classNames( 'plans-features-main__notice', {
+						'is-wpcom':
+							! isJetpack && ! isInSignup && ! isLandingPage && ! isLaunchPage ? 'is-wpcom' : '',
```
Does the value need to be a string? I think I've only seen booleans
```
{ 'is-wpcom': ! isJetpack && ! isInSignup && ! isLandingPage && ! isLaunchPage  }
```

Commit:
set is-wpcom className as boolean

Commit:
set is-wpcom className as boolean

dzver reviewed (commented):
Something with the margins is off if I don't have credits but do have the plan:

<img width="1074" alt="Screenshot 2022-07-19 at 14 52 47" src="https://user-images.githubusercontent.com/82778/179744512-5b1a63e8-83e9-4752-9d9d-a6ab835d707d.png">

worldomonation commented on test/e2e/specs/plans/plans__legacy-renew.ts:
```diff
@@ -44,12 +43,6 @@ describe( DataHelper.createSuiteTitle( 'Plans (Legacy): Renew' ), function () {
 		it( `${ planName } is the active plan`, async function () {
 			await plansPage.validateActivePlan( planTier );
 		} );
-
-		it( 'Legacy plan notice is shown', async function () {
-			const noticeComponent = new NoticeComponent( page );
-			const message = `You’re currently on a legacy plan. If you’d like to learn about your eligibility to switch to a Pro plan please contact support.`;
-			await noticeComponent.noticeShown( message );
-		} );
```
I think I saw a similar changeset on another one of the revert PRs. Might want to make sure this is actually necessary.

ciprianimike replied:
Thanks for letting me know @worldomonation, I'll rebase and double check it

ciprianimike committed:
Move credits notice into plans-features-main container.

bring notices elements to the same level to make possible use the CSS + operator

ciprianimike committed:
add mobile breakpoint to CSS

ciprianimike committed:
solve conflict rebase trunk on client/my-sites/site-settings/form-general.jsx

ciprianimike committed:
solve conflict rebase trunk on client/my-sites/plans-features-main/index.jsx

ciprianimike commented:
@dzver The feature branch was rebased and is now clean, the conflicts were solved and E2E tests now pass since they were fixed on trunk.

Plans tested:
- Free
- Personal
- Starter
- Premium
- Pro
- Business
- eCommerce (Atomic)

### No notices
Status | Mobile | Table | Desktop
-- | -- | -- | --
before | <img width="414" alt="1 - before-free_plan-no_notice-mobile" src="https://user-images.githubusercontent.com/5706607/180786629-b4a7b58e-faa0-4332-943d-f9b8703f61da.png"> | <img width="657" alt="2 - before-free_plan-no_notice_tablet" src="https://user-images.githubusercontent.com/5706607/180786676-fc51632a-b164-4c4a-90ee-a7dd5627c520.png"> | <img width="1139" alt="3 - before-free_plan-no_notice-desktop" src="https://user-images.githubusercontent.com/5706607/180786735-e6b07bc3-82b2-4f4b-83ac-92cbba9bf5e8.png">
after | <img width="414" alt="1 - after-free_plan-no_notice-mobile" src="https://user-images.githubusercontent.com/5706607/180786828-7329dcba-f16d-4d6d-ad58-2d6601b74e77.png"> | <img width="657" alt="2 - after-free_plan-no_notice_tablet" src="https://user-images.githubusercontent.com/5706607/180786894-db7b6790-d9b4-4863-aa47-95134fd3d082.png"> | <img width="1139" alt="3 - before-free_plan-no_notice-desktop" src="https://user-images.githubusercontent.com/5706607/180786929-30112fdf-78fe-42cc-9fcf-64d54336cd88.png">

### 1 Notice
Status | Mobile | Table | Desktop
-- | -- | -- | --
before | <img width="414" alt="1 - before-premium_plan-1-notice-mobile" src="https://user-images.githubusercontent.com/5706607/180790221-853ec653-2cd9-409a-97dd-252d8f428ad6.png"> | <img width="816" alt="2 - before-premium_plan-1-notice-tablet" src="https://user-images.githubusercontent.com/5706607/180790268-74bec578-108b-4587-abd2-e64f4471fafa.png"> | <img width="1599" alt="3 - before-premium_plan-1-notice-desktop" src="https://user-images.githubusercontent.com/5706607/180790540-4e814c86-940c-4161-8e04-a3826346988c.png">
after | <img width="413" alt="1 - after-premium_plan-1-notice-mobile" src="https://user-images.githubusercontent.com/5706607/180790444-748a467c-3e6b-4f65-9693-226f3e666a04.png"> | <img width="820" alt="2 - after-premium_plan-1-notice-tablet" src="https://user-images.githubusercontent.com/5706607/180790500-7f2da2bf-572f-4752-b5f9-1c8ccaa54d21.png"> | <img width="1603" alt="3 - after-premium_plan-1-notice-desktop" src="https://user-images.githubusercontent.com/5706607/180790313-d79cca15-7a07-48d5-ace4-49d15175985e.png">

### 2 Notices
Status | Mobile | Table | Desktop
-- | -- | -- | --
before | <img width="391" alt="1 - before-pro_plan-2-notices-mobile" src="https://user-images.githubusercontent.com/5706607/180790888-8e230936-f57d-4b0d-a9ec-d390d1793ffa.png"> | <img width="818" alt="2 - before-pro_plan-2-notices-tablet" src="https://user-images.githubusercontent.com/5706607/180790912-4919eeaf-e612-4fd3-b022-b57f4309ed80.png"> | <img width="1597" alt="3 - before-pro_plan-2-notices-desktop" src="https://user-images.githubusercontent.com/5706607/180790951-0740be75-7676-408a-95a5-6bf614f50303.png">
after | <img width="413" alt="1 - after-pro_plan-2-notices-mobile" src="https://user-images.githubusercontent.com/5706607/180791083-3c74d1cf-a582-40fa-b858-798cd09e19a6.png"> | <img width="819" alt="2 - after-pro_plan-2-notices-tablet" src="https://user-images.githubusercontent.com/5706607/180791310-dedd4a9b-905a-470a-8686-599867be59ad.png"> | <img width="1599" alt="3 - after-pro_plan-2-notices-desktop" src="https://user-images.githubusercontent.com/5706607/180791258-f323445e-1a65-407e-9f98-4bd9dd39e687.png">

wongasy commented:
@ciprianimike I discovered this old PR while playing with the code review reminder scheduler feature :) Is this PR still needed? If not, shall we close it?
